lfsapi/auth.go

```diff
@@ -24,28 +24,38 @@ var (
 // DoWithAuth sends an HTTP request to get an HTTP response. It attempts to add
 // authentication from netrc or git's credential helpers if necessary,
 // supporting basic and ntlm authentication.
-func (c *Client) DoWithAuth(remote string, req *http.Request) (*http.Response, error) {
-	return c.doWithAuth(remote, req, nil)
+func (c *Client) DoWithAuth(remote string, access Access, req *http.Request) (*http.Response, error) {
+	return c.doWithAuth(remote, access, req, nil)
 }
 
-func (c *Client) doWithAuth(remote string, req *http.Request, via []*http.Request) (*http.Response, error) {
+// DoAPIRequestWithAuth sends an HTTP request to get an HTTP response similarly
+// to DoWithAuth, but using the LFS API endpoint for the provided remote and
+// operation to determine the access mode.
+func (c *Client) DoAPIRequestWithAuth(remote string, req *http.Request) (*http.Response, error) {
+	operation := getReqOperation(req)
+	apiEndpoint := c.Endpoints.Endpoint(operation, remote)
+	access := c.Endpoints.AccessFor(apiEndpoint.Url)
+	return c.doWithAuth(remote, access, req, nil)
```

Comment:
I had a comment here that I wrote about halfway during the review that recommended that you call `DoWithAuth()` here to minimize any diff in the future should we complicate the body of `DoWithAuth()` at all later on.

But, you made that change in the subsequent commit, so I removed that comment here :-). Instead, I would recommend that in the future you make that change here (i.e., introduce this new function with a call to `DoWithAuth()` in this commit) rather than introduce something and then immediately change it in a future commit.